cli/command/container/opts.go

```diff
@@ -816,9 +816,6 @@ func parseNetworkAttachmentOpt(ep opts.NetworkAttachmentOpts) (*networktypes.End
 		if len(ep.Aliases) > 0 {
 			return nil, errors.New("network-scoped aliases are only supported for user-defined networks")
 		}
-		if len(ep.Links) > 0 {
-			return nil, errors.New("links are only supported for user-defined networks")
-		}
```

Comment:
I dug up my original PR / commit that changed this code; https://github.com/docker/cli/commit/5bc09639cc0aa382fdfcd9ca94d7817c8b3073c0

A couple of things there that I was curious about (but it's a long time); on this line, there's an error that treats `container.links` and `ep.links` as separate things; https://github.com/docker/cli/commit/5bc09639cc0aa382fdfcd9ca94d7817c8b3073c0#diff-e546b85d27f9dbea4fd41e83a58e36bb540bcda3e304355d56c92a119cd5aa2aR721

```go
	if len(n.Links) > 0 && copts.links.Len() > 0 {
		return errdefs.InvalidParameter(errors.New("conflicting options: cannot specify both --link and per-network links"))
	}
```



It's valid to use `--link` on the default ("bridge") network, but for that network (currently) only with the legacy link implementation (hard-wired links). 

Do you know if the daemon-side code makes this distinction as well when using the network-config make / slice? (i.e. does it look "if its the `default` (bridge) network, then use legacy links, and otherwise the DNS-based links?) or would the daemon only look at `container.links` to create legacy links?

Reply:
note that the above validation is only performed on the first network in the list (which holds data of the "default" network)

Reply:
You're right:

* `copts.links` will be put into `HostConfig.Links`, which is the way to use legacy links;
* `n.Links` is used for DNS-based links.

I removed the first commit (ie. the one removing this check) and instead I changed the condition under which `copts.links` is copied into `n.Links` (ie. to not do that on "default" network).